core/internal/watcher/watcher.go

```diff
@@ -14,35 +15,100 @@ const pollingInterval = time.Millisecond * 100
 
 type Watcher struct {
 	watcher  *fw.Watcher
-	wg       *sync.WaitGroup
 	logger   *observability.CoreLogger
 	registry *registry
+	filesDir string
+	started  *atomic.Bool
 }
 
-type WatcherOption func(*Watcher)
+type Params struct {
+	Logger *observability.CoreLogger
 
-func WithLogger(logger *observability.CoreLogger) WatcherOption {
-	return func(w *Watcher) {
-		w.logger = logger
-	}
+	// The directory where files are stored.
+	//
+	// Used as the base for relative file paths.
+	FilesDir string
 }
 
-func New(opts ...WatcherOption) *Watcher {
-	w := &Watcher{
+func New(params Params) *Watcher {
+	logger := params.Logger
+	if logger == nil {
+		logger = observability.NewNoOpLogger()
+	}
```

Comment:
do you think we should make it the default pattern everywhere we are expecting a logger?

Reply:
yeah I think that would be good. It's very easy to get a panic by leaving it nil